tests/integration/components/tasks/extension-form-test.js

```diff
@@ -0,0 +1,186 @@
+      server.create('extensionRequest', obj);
+    });
+  });
+
+  const toggleExtensionForm = (setter, name) => {
+    extensionFormOpened = !extensionFormOpened;
+    if (setter) {
+      setter(name, extensionFormOpened);
+    }
+  };
+
+  const closeExtensionForm = (setter, name) => () => {
+    extensionFormOpened = false;
+    if (setter) {
+      setter(name, false);
+    }
+  };
+
+  const closeExtensionModel = (setter, name) => (e) => {
+    if (!e) {
+      toggleExtensionForm(setter, name);
+      return;
+    }
+    e.stopPropagation();
+    if (e.target.classList.contains('extension-form__container-back')) {
+      toggleExtensionForm(setter, name);
+    }
+  };
+
+  test('When Clicked /"Close Form/" button or background task extension form should unmount', async function (assert) {
+    this.set('task', tasksData[0]);
+    this.set(
+      'closeExtensionModel',
+      closeExtensionModel(this.set, 'extensionFormOpened')
+    );
+    this.set(
+      'closeExtensionForm',
+      closeExtensionForm(this.set, 'extensionFormOpened')
+    );
+    this.set('extensionFormOpened', extensionFormOpened);
+
+    await render(
+      hbs`
+      {{#if this.extensionFormOpened}}
+      <Task::ExtensionForm
+      @task={{this.task}}
+      @closeForm={{this.closeExtensionForm}}
+      @title='Form for extension Request'
+      @closeModel={{this.closeExtensionModel}}
+      />
+      {{/if}}`
+    );
+    assert
+      .dom(this.element.querySelector('[data-test-title]'))
+      .hasText('Form for extension Request');
+    await click(this.element.querySelector('.extension-form__container-close'));
+    assert.dom(this.element.querySelector('[data-test-title]')).doesNotExist();
+
+    extensionFormOpened = true;
+    this.set('extensionFormOpened', extensionFormOpened);
+    assert
+      .dom(this.element.querySelector('[data-test-title]'))
+      .hasText('Form for extension Request');
+    await click(this.element.querySelector('.extension-form__container-back'));
+    assert.dom(this.element.querySelector('[data-test-title]')).doesNotExist();
+  });
+
+  test('When no extension requests found, the option to create extension request should show and then later open form', async function (assert) {
+    this.set('task', tasksData[3]);
+    this.set(
+      'closeExtensionModel',
+      closeExtensionModel(this.set, 'extensionFormOpened')
+    );
+    this.set(
+      'closeExtensionForm',
+      closeExtensionForm(this.set, 'extensionFormOpened')
+    );
+
+    await render(
+      hbs`<Task::ExtensionForm @task={{this.task}} @closeForm={{this.closeExtensionForm}} @title='Form for extension Request' @closeModel={{this.closeExtensionModel}}/>`
+    );
+    await waitFor('.extension-form__open-button');
+    assert
+      .dom(this.element.querySelector('.extension-form__open-button'))
+      .hasText('Create an extension request');
+
+    await click(this.element.querySelector('.extension-form__open-button'));
```

Comment:
Please use `data-test-selectors` to get the elements
Guide Reference :
https://guides.emberjs.com/release/testing/testing-tools/#toc_ember-test-selectors